src/scripts/version_history.sh

```diff
@@ -3,20 +3,32 @@
 # ensure consequences still work if this script blows up
 touch .deployment_changelog
 
-if [ "$K8S_DEPLOYMENT_NAME" == "PROJECT_NAME" ]; then
-  K8S_DEPLOYMENT_NAME="$CIRCLE_PROJECT_REPONAME"
+if [ "${K8S_DEPLOYMENT_NAME}" == "PROJECT_NAME" ]; then
+  K8S_DEPLOYMENT_NAME="${CIRCLE_PROJECT_REPONAME}"
 fi
 
+current_commit="$(echo "${APP_VERSION}" | cut -d'.' -f3)"
+
+# Try and get the currently deployed version
+K8S_PREVIOUS_APP_VERSION="$(kubectl get "deployment/${K8S_DEPLOYMENT_NAME}" --namespace="${KUBE_ENV_NAMESPACE}" -o=jsonpath='{.metadata.labels.app\.kubernetes\.io/version}' || true)"
+
 if [ "$K8S_PREVIOUS_APP_VERSION" == "" ]; then
-  K8S_PREVIOUS_APP_VERSION="$(kubectl get "deployment/$K8S_DEPLOYMENT_NAME" --namespace="$KUBE_ENV_NAMESPACE" -o=jsonpath='{.metadata.labels.app\.kubernetes\.io/version}')"
+  # if no previous version was found, set to current commit minus 1
+  echo "Previous deployment not found, showing current commit only." >> .deployment_changelog
+  previous_commit="${current_commit}^1"
```

Comment:
📝 this works on CircleCI, just want to flag it depends on the CI if this history is available
eg. github actions don't pull down anything apart from the exact commit

Reply:
To me this is an message case, as the deployment notification already contains the currently deployed SHA